internal/internal_activity.go

```diff
@@ -432,6 +432,10 @@ func deSerializeFunctionResult(f interface{}, result []byte, to interface{}, dat
 	}
 
 	// For everything we return result.
+	// Code reaches here for 2 cases:
+	// 	1. activity is executed by name (not the func pointer) and it wasn't registered
+	// 	2. activity is executed by func pointer and the signature indicates it doesn't/can't return data.
+	//		for example it only has one return parameter (which can only be be error).
 	return decodeArg(dataConverter, result, to)
 }
```

Comment:
Huh.  On closer look, this whole chain of code is pretty strange:
- `deSerializeFunctionResult` is only called if `result` is non-nil
- `deSerializeFnResultFromFnType` is only called by this func, and it does nothing if `result` is nil
- none of this needs the _registered func type_ at all:
  - if everything agrees, the type does nothing to contribute to the behavior
  - if there if there is data + an out param + the registered activity _does not_ return data, the earlier code silently does nothing (the data exists but is inaccessible)
  - if it's _not_ registered, it falls back to just decoding directly, which is what `deSerializeFnResultFromFnType` does anyway

... I think `deSerializeFunctionResult` might be more-correctly replaced with just a call to `decodeArg` instead.

Reply:
TBH I need to re-debug this to remember how I ended up finding those 2 cases. Maybe I meant to add this somewhere else. From the code in this function it seems 2 is not possible.